common/chains/client/types/rpc_client.go

```diff
@@ -0,0 +1,83 @@
+package types
+
+import (
+	"context"
+	"math/big"
+
+	feetypes "github.com/smartcontractkit/chainlink/v2/common/fee/types"
+	"github.com/smartcontractkit/chainlink/v2/common/types"
+	"github.com/smartcontractkit/chainlink/v2/core/assets"
+)
+
+type RPCClient[
+	CHAIN_ID types.ID,
+	SEQ types.Sequence,
+	ADDR types.Hashable,
+	BLOCK_HASH types.Hashable,
+	TX any,
+	TX_HASH types.Hashable,
+	EVENT any,
+	EVENT_OPS any, // event filter query options
+	TX_RECEIPT types.Receipt[TX_HASH, BLOCK_HASH],
+	FEE feetypes.Fee,
+	HEAD types.Head[BLOCK_HASH],
+] interface {
+	Accounts[ADDR, SEQ]
+	Transactions[ADDR, TX, TX_HASH, BLOCK_HASH, TX_RECEIPT, SEQ, FEE]
+	Events[EVENT, EVENT_OPS]
+	Blocks[HEAD, BLOCK_HASH]
+
+	BatchCallContext(ctx context.Context, b []any) error
+	CallContract(
+		ctx context.Context,
+		attempt interface{},
+		blockNumber *big.Int,
+	) (rpcErr []byte, extractErr error)
+	CallContext(ctx context.Context, result interface{}, method string, args ...interface{}) error
+	ChainID(ctx context.Context) (CHAIN_ID, error)
+	CodeAt(ctx context.Context, account ADDR, blockNumber *big.Int) ([]byte, error)
+
+	Subscribe(ctx context.Context, channel chan<- HEAD, args ...interface{}) (types.Subscription, error)
+}
+
+type Accounts[ADDR types.Hashable, SEQ types.Sequence] interface {
```

Comment:
Would it make sense to name these sub-interfaces as `*Client`s also?
```suggestion
type AccountClient[ADDR types.Hashable, SEQ types.Sequence] interface {
```

Reply:
I think for now if these types are only used by the `RPCClient` for better readability, it's simpler to keep them as they are.

Reply:
Well, I was under the impression that these subinterfaces were created simply for housekeeping, so it made sense to me that if they were going to be kept under a generalized `RPCClient` then the client suffix seemed a bit redundant.

Reply:
But they aren't a property of the `RPCClient` type - they exist on their own and that type is defined by referencing them.

If the `*Client` would be redundant, doesn't that also suggest that `RPCClient` is redundant too? Since we are already in `client/types`? What about s/`RPCClient`/`RPC`?

`RPC` still feels out of place to me in any case, since they are all RPC client types - would finding a better term for this help?

Reply:
Maybe just s/`RPCClient`/`Chain` would be clear about meaning 'everything'?
I wish we didn't need an extra `package types` - they always confuse the naming. It would be nice to have `client.Chain` or whatever with the existing `client.Accounts`, `client.Transactions`, etc.

Reply:
> I would prefer s/RPCClient/RPC, since RPC name is more relatable to anyone.
Chain is an overloaded term, and I find it more confusing than RPC.

But then why don't the other interfaces need to say that they are RPC?
Also, the interface is general enough to non-RPC implementations.

> But we do need to separate a package's exported interfaces, from its implementations. This avoids circular dependencies.

Is that necessarily true in this specific case? That generally only happens when there are other unnecessary constraints.

Reply:
I'm not opposed into simplifying the client suffixes, but we have the following problem: there are multiple interfaces under the same package that require different client i.e. chainclient and nodeclient. To distinguish those we would have to create new packages or add the client suffix.

Reply:
Also, we could entirely remove the Accounts, Transactions, etc. interfaces since they aren't used anywhere besides RPCClient, and just add a comment.